src/NuGetGallery/Infrastructure/AddMigrateToOrganization.Up.sql

```
@@ -0,0 +1,47 @@
+﻿-- Copyright (c) .NET Foundation. All rights reserved.
+-- Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+SET ANSI_NULLS ON
+
+IF OBJECT_ID('[dbo].[MigrateToOrganization]', 'P') IS NOT NULL
+    DROP PROCEDURE  [dbo].[MigrateToOrganization]
+GO
+
+CREATE PROCEDURE [dbo].[MigrateToOrganization]
+(
+	@orgKey INT,
+	@adminKey INT,
+	@token NVARCHAR(MAX)
+)
+AS
+BEGIN
+	DECLARE @reqCount INT
+
+	-- Ensure migration request exists
+	SELECT @reqCount = COUNT(*)
+	FROM [dbo].[OrganizationMigrationRequests]
+	WHERE NewOrganizationKey = @orgKey
+		AND AdminUserKey = @adminKey
+		AND ConfirmationToken = @token
+	IF @reqCount = 0 RETURN
+
+	BEGIN TRANSACTION
+	BEGIN TRY
+		-- Ensure Organizations do not have credentials or memberships
```

This isn't "ensuring", this is wiping any of these existing entities if they exist.

IMO we should fail the migration if the user has any memberships--that would suggest that the user was already an organization, which is not something we should just ignore.

---

Sounds reasonable... I'll add a SELECT outside the transaction and fail fast if the account has any memberships.

Gallery-side, we should also check for memberships before allowing migration. Account would need to be removed from organization(s) to be eligible for migration.